lib/inputs/base-input.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+class BaseInput {
+    constructor(type, id, params, inputDefaultValue) {
+        this.type = type;
+        this.id = id;
+        this.params = this.normalizeParams(params || {});
+
+        this.value = this.params.default || this.getInputDefaultValue();
+    }
+
+    getInputDefaultValue() {
+        throw new Error('getInputDefaultValue must be implemented');
+    }
+
+    normalizeParams(params) {
+        return params || {};
```

Comment:
Same thing as in the constructor — is there any case when `params` wouldn’t be set? If not, I’d just make it a required parameter.

Here, the issue is even more pronounced as a method called `normalizeParams` really shouldn’t invent its own params when not supplied any.